libtock/nrf51_serialization.c

```diff
@@ -2,27 +2,56 @@
 
 int nrf51_serialization_reset (void) {
   // Reset the nRF51 chip
-  return command(DRIVER_NUM_NRF_SERIALIZATION, 2, 0, 0);
+  syscall_return_t sval = command2(DRIVER_NUM_NRF_SERIALIZATION, 3, 0, 0);
+  if (sval.type == TOCK_SYSCALL_SUCCESS) {
+    return TOCK_SUCCESS;
+  } else {
+    return TOCK_FAIL;
+  }
 }
 
 int nrf51_serialization_subscribe (subscribe_cb cb) {
   // get some callback love
-  return subscribe(DRIVER_NUM_NRF_SERIALIZATION, 0, cb, NULL);
+  subscribe_return_t sval = subscribe2(DRIVER_NUM_NRF_SERIALIZATION, 0, cb, NULL);
+  if (!sval.success) {
+    return tock_error_to_rcode(sval.error);
+  }
+
+  // Instruct the capsule to start receiving. Doesn't
+  syscall_return_t cval = command2(DRIVER_NUM_NRF_SERIALIZATION, 2, 0, 0);
+  if (cval.type == TOCK_SYSCALL_SUCCESS_U32) {
```

Comment:
sort of odd that this command returns a `success_u32` if userspace doesn't even need the value

Reply:
The userspace library API is pretty coupled to the semantics of the nRF stack, which is typical embedded code (makes lots of assumptions).

Reply:
It is a bit weird that a call to subscribe also makes a call to read; this isn't the behavior that I would expect. I'd rather not have this weird wart of non-standard behavior. #174 fixes `serialization_tock.c` to correctly invoke a read operation.